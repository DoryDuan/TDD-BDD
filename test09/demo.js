export const generateConfig = () => {
  return {
    server: "http://localhost",
    port: 8000,
    domain: "localhost1",
    time: new Date(),
  };
};

export const generateConfig1 = () => {
  return {
    server: "http://localhost2",
    port: 8000,
    domain: "localhost",
    time: new Date(),
  };
};
