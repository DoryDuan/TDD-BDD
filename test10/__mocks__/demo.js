// 测试的时候，用这个假的 getData 替换被测试文件中真的 getData
export const getData = () => {
  return new Promise((resolve, reject) => {
    resolve("hello");
  });
};
