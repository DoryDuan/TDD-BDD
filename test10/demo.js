import axios from "axios";

export const getData = () => {
  return axios.get("/api").then((res) => res.data);
};

export const getNumber = () => {
  return 123;
};
