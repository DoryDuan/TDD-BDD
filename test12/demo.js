import Util from "./util";
const demoFunc = (x, y) => {
  const util = new Util();
  util.a(x);
  util.b(y);
};

export default demoFunc;
