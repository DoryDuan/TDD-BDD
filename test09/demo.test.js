import { generateConfig, generateConfig1 } from "./demo";

// 适合配置文件和组件测试
// 第一次会生成快照文件，只要没有修改，快照测试以后就都能通过
// 快照测试发生改变,可以按 u 来更新快照测试文件
// 多个快照测试发生改变，可以按 i 来一个一个更新快照测试,然后根据提示按 u,s,q等
// 配置文件 time 是一个一直改变的数据，可以通过给 toMatchSnapshot 传参让测试通过
test("测试 generateConfig", () => {
  expect(generateConfig()).toMatchSnapshot({
    time: expect.any(Date),
  });
});

test("测试 generateConfig1", () => {
  expect(generateConfig1()).toMatchSnapshot({
    time: expect.any(Date),
  });
});

// 行内 snapshot，把快照放到测试用例里面
// 需要安装模块 cnpm install prettier@1.18.2 -D
test("测试 generateConfig1", () => {
  expect(generateConfig1()).toMatchInlineSnapshot(
    {
      time: expect.any(Date),
    },
    `
    Object {
      "domain": "localhost",
      "port": 8000,
      "server": "http://localhost2",
      "time": Any<Date>,
    }
  `
  );
});
