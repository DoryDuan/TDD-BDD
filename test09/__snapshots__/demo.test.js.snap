// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateConfig 1`] = `
Object {
  "domain": "localhost1",
  "port": 8000,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;

exports[`测试 generateConfig1 1`] = `
Object {
  "domain": "localhost",
  "port": 8000,
  "server": "http://localhost2",
  "time": Any<Date>,
}
`;
